release/models/system/openconfig-hashing.yang

```diff
@@ -0,0 +1,326 @@
+                      "Operational state data at the global hash policy
+                      level.";
+
+                    uses hashing-policy-config;
+                }
+                container hash-field-modes {
+                    description
+                        "Container for specifying inputs to be used when
+                        calculating the hash.";
+                    container config {
+                        description
+                            "Configurable items at the hashing inputs level.";
+                        uses hashing-inputs;
+                    }
+                    container state {
+                        config false;
+                        description "Operational state data at the hashing
+                            inputs level.";
+                        uses hashing-inputs;
+                    }
+                    container ipv4-fields {
+                        description "ipv4 hashing input fields.";
+                        uses ipv4-fields-top;
+                    }
+                    container ipv6-fields {
+                        description "ipv6 hashing input fields.";
+                        uses ipv6-fields-top;
+                    }
+                }
+            }
+        }
+    }
+
+    grouping supported-algorithms-state {
+        description
+            "Top-level container for the supported algorithms.";
+        leaf name {
+            type string;
+            description
+                "Name of the supported algorithm.";
+        }
+        leaf description {
+            type string;
+            description
+                "Description of the supported algorithm.";
+        }
+    }
+    grouping supported-algorithms-top {
+        description
+            "Top-level container of the supported algorithms for supported
+            algorithms.  The list of algorithms are expected to be defined by
+            the target device.";
+        container supported-algorithms {
+            config false;
+            description
+                "Container of the supported algorithms for supported
+                algorithms.";
+            list supported-algorithm {
+                key "name";
+
+                description
+                    "List of the supported algorithms for supported
+                    algorithms.";
+
+                leaf name {
+                    type leafref {
+                        path "../state/name";
+                    }
+                    description
+                        "Reference to the supported algorithm list key.";
+                }
+
+                container state {
+                    description
+                        "Operational data for the supported algorithm.";
+                    config false;
+                    uses supported-algorithms-state;
+                }
+            }
+        }
+
+    }
+
+    uses supported-algorithms-top;
+    uses hashing-top;
```

Comment:
replace this with augment statements.  This way hashing adds to  the /system model, while staying independent.  One can choose to implement the hashing model or not without affecting the /system model.

Example:
```
  augment "/oc-system:system" {
    description
      "Hashing algorithms and hashing policies";

    uses supported-algorithms-top;
    uses hashing-top;
  }
```